packages/thirdweb/src/utils/signatures/helpers/parseTypedData.ts

```diff
@@ -0,0 +1,19 @@
+import type { TypedData } from "abitype";
+import type { TypedDataDefinition } from "viem";
+import { isHex } from "../../encoding/hex.js";
+
+/**
+ * @internal
+ */
+export function parseTypedData<
+  typedData extends TypedData | Record<string, unknown> = TypedData,
+  primaryType extends keyof typedData | "EIP712Domain" = keyof typedData,
+>(
+  typedData: TypedDataDefinition<typedData, primaryType>,
+): TypedDataDefinition<typedData, primaryType> {
+  const domain = typedData.domain as unknown & { chainId?: unknown }; // TODO: create our own typed data types so this is cleaner
+  if (domain?.chainId !== undefined && isHex(domain.chainId)) {
+    throw new Error("signTypedData: chainId must be of type number");
```

Comment:
can we not transform it ourselves to a number?

Reply:
We can't because this type is _very_ particular and it's a combination of viem types and abitype so there's no way to get around it until we create our own from the ground up (which we should do just afraid to try it now)

Reply:
ok